### DISCUSSION TITLE
Need tutorials

### DISCUSSION THREAD

**Discussion by ianzone** [dynamoose/dynamoose#1473] [Q&A] [answered]
Hi there, I'm new to dynamoose, I hope there could be a tutorial to teach me the lifecycle of using dynamoose, from creating tables and CURD actions to altering schemas and data migration. Like the tutorial described here https://documentation.divio.com/ . Many thanks.

**Comment by fishcharlie** [comment, marked as answer]
Hello @ianzone. Thanks for checking out Dynamoose. I understand the complexity of getting started with Dynamoose. However, we have a couple of great resources that already exist for you:

**Introduction to Dynamoose**

This is a course that I personally wrote, recorded, produced, and edited going over the basics of Dynamoose. It contains just over an hour of video lessons describing the basic features of Dynamoose. This course is hosted through my companies rrainn Learn product. You can purchase life time access for $9.99 at [this link](https://learn.rrainn.com/courses/dd4dc0b8-f97d-4a77-9ce5-337d9bb15771).

**[DynamooseJS.com](https://DynamooseJS.com)**

Dynamoose's website is a great resource that contains information about each method and property available in the API. The documentation also includes sample code, and other information about the package.

**Sponsor Tiers**

I also have [GitHub Sponsor tiers](https://github.com/sponsors/fishcharlie) setup, some of which include private calls to help explain anything you wish.

**Resources Directory**

We also have a [resources repository](https://github.com/dynamoose/resources) that users can contribute new Dynamoose resources to.

**Stack Overflow**

Although not necessarily tutorial based, we do have a [tag on Stack Overflow](https://stackoverflow.com/questions/tagged/dynamoose) that you can ask questions on.

**Slack**

Again, not necessarily tutorial based, but we also have a [Slack domain](https://join.slack.com/t/dynamoose/shared_invite/enQtODM4OTI0MTc1NDc3LWI3MmNhMThmNmJmZDk5MmUxOTZmMGEwNGQzNTRkMjhjZGJlNGM5M2JmZjMzMzlkODRhMGY3MTQ5YjQ2Nzg3YTY) that you can join.

**Source Code**

Finally, all of the Dynamoose source code is public and open source.

---

Overtime I do hope to continue to improve the Dynamoose documentation, provide more examples, as well as develop more courses and tutorials to help the community.

However, it is important to keep in mind that Dynamoose is an open source project. My time is valuable, and that leads to the need to prioritize objectives and goals within the project, which is why a couple of those are paid options. It's rare to see 1st party free tutorials for open source projects. Most open source projects simply provide documentation. Sometimes community based tutorials and resources are developed, however most of those options are normally paid, or include advertising. I currently don't have a platform to setup advertising on the work I develop.

Not saying more free courses and tutorials won't be developed overtime, but it is a balance that I do keep in mind.

---

I hope this helps!
